### PR TITLE
system-variables: update from generated source

### PR DESCRIPTION
**Pull request by morgo**
### What is changed, added or deleted? (Required)

This updates system variables from a script (against tidb@master) to guarantee it is accurate.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -254,11 +254,6 @@ mysql> SELECT * FROM t1;
 - Default value: ""
 - This variable is used to specify a list of storage engines that might fall back to TiKV. If the execution of a SQL statement fails due to a failure of the specified storage engine in the list, TiDB retries executing this SQL statement with TiKV. This variable can be set to "" or "tiflash". When this variable is set to "tiflash", if the execution of a SQL statement fails due to a failure of TiFlash, TiDB retries executing this SQL statement with TiKV.
 
-### tidb_allow_function_for_expression_index <span class="version-mark">New in v5.2.0</span>
-
-- Scope: NONE
-- This variable is used to show the functions that are allowed to be used for creating expression indexes.
-
```
Comment (morgo) [inline system-variables.md:261]: @wjhuang2016 This variable no longer appears to be in tidb@master. What's the backstory, was it removed?

File: system-variables.md [hunk 2]
```diff
@@ -1201,7 +1196,7 @@ SET tidb_query_log_max_len = 20
 
 ### tidb_replica_read <span class="version-mark">New in v4.0</span>
 
-- Scope: SESSION
+- Scope: SESSION | GLOBAL
```
Comment (morgo) [inline system-variables.md:1199]: From https://github.com/pingcap/tidb/pull/27622

**Review by dveeden — APPROVED** [review]

File: system-variables.md (hunk 1 above)
Comment (morgo) [reply system-variables.md:261]: I confirmed with @wjhuang2016 that it is not in master, but it is in 5.2, and will be added to 5.3 before release. So I am going to add it back in now for simplicity.

**Review by TomShawn — APPROVED** [review]

**Pull request merged**